tokio-util/src/io/sync_bridge.rs

```diff
@@ -6,6 +6,81 @@ use tokio::io::{
 
 /// Use a [`tokio::io::AsyncRead`] synchronously as a [`std::io::Read`] or
 /// a [`tokio::io::AsyncWrite`] as a [`std::io::Write`].
+///
+/// # Alternatives
+///
+/// In many cases, there are better alternatives to using `SyncIoBridge`, especially if you
+/// want to avoid blocking the async runtime. Consider the following scenarios:
+///
+/// ## Example 1: Hashing Data
+///
+/// When hashing data, using `SyncIoBridge` can lead to suboptimal performance and might not fully leverage the async capabilities of the system.
```

Comment:
It would be nice if this explained _why_ using `SyncIoBridge`  "can lead to suboptimal performance and might not fully leverage the async capabilities of the system"; I think just saying "you might have bad performance" is not as helpful as explaining the specific details, so that the user can know what issues they are avoiding and why they matter?